sbpy/bib/core.py

```diff
@@ -29,8 +29,26 @@ def register(task, citations):
     """
     global _bibliography, _track
     if _track:
-        _bibliography[task] = _bibliography.get(task, {})
-        _bibliography[task].update(citations)
+        if task in _bibliography:
+            key = []
```

Comment:
I think you don't have to define `key` and `value` if you are using the loop I suggest below.